sdk/packages/react-next/src/hooks/useParseOpenEvent.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  type ParseOpenEventError,
+  type ParseOpenEventReturn,
+  parseOpenEvent,
+} from '@omni-network/core'
+import { useMemo } from 'react'
+import type { Log } from 'viem'
+import type { UseWaitForTransactionReceiptReturnType } from 'wagmi'
+
+type UseParseOpenEventParams = {
+  status: UseWaitForTransactionReceiptReturnType['status']
+  logs?: Log[]
+}
+
+type UseParseOpenEventReturn = {
+  resolvedOrder: ParseOpenEventReturn | undefined
+  error: ParseOpenEventError | undefined
+}
```

Comment:
type narrowing?

Reply:
I reused the type as currently used by the react package, but yes I can change it if you want?
To me for this kind of pattern it would be best to use something like [typescript-result](https://github.com/everweij/typescript-result) for example, rather than ad-hoc types every time.